tests/ZonedDateTimeTest.php

```diff
@@ -814,24 +814,38 @@ public function providerToNativeDateTime(): array
         ];
     }
 
-    public function testJsonSerialize(): void
+    /** @dataProvider providerToString */
+    public function testJsonSerialize(string $localDateTime, string $timeZoneIso, string $expectedString): void
     {
-        $timeZone = TimeZone::parse('America/Los_Angeles');
-        $localDateTime = '2000-01-20T12:34:56.123456789';
         $localDateTime = LocalDateTime::parse($localDateTime);
-        $zonedDateTime = ZonedDateTime::of($localDateTime, $timeZone);
+        $zonedDateTime = ZonedDateTime::of($localDateTime, TimeZone::parse($timeZoneIso));
 
-        self::assertSame(json_encode('2000-01-20T12:34:56.123456789-08:00[America/Los_Angeles]'), json_encode($zonedDateTime));
+        self::assertSame(json_encode($expectedString), json_encode($zonedDateTime));
     }
 
-    public function testToString(): void
+    /** @dataProvider providerToString */
+    public function testToISOString(string $localDateTime, string $timeZoneIso, string $expectedString): void
     {
-        $timeZone = TimeZone::parse('America/Los_Angeles');
-        $localDateTime = '2000-01-20T12:34:56.123456789';
         $localDateTime = LocalDateTime::parse($localDateTime);
-        $zonedDateTime = ZonedDateTime::of($localDateTime, $timeZone);
+        $zonedDateTime = ZonedDateTime::of($localDateTime, TimeZone::parse($timeZoneIso));
+
+        self::assertSame($expectedString, $zonedDateTime->toISOString());
+    }
 
-        self::assertSame('2000-01-20T12:34:56.123456789-08:00[America/Los_Angeles]', (string) $zonedDateTime);
+    /** @dataProvider providerToString */
+    public function testToString(string $localDateTime, string $timeZoneIso, string $expectedString): void
+    {
+        $localDateTime = LocalDateTime::parse($localDateTime);
+        $zonedDateTime = ZonedDateTime::of($localDateTime, TimeZone::parse($timeZoneIso));
+
+        self::assertSame($expectedString, (string) $zonedDateTime);
+    }
+
+    public function providerToString(): array
+    {
+        return [
+            ['2000-01-20T12:34:56.123456789', 'America/Los_Angeles', '2000-01-20T12:34:56.123456789-08:00[America/Los_Angeles]'],
```

Comment:
Could you please add another one with an offset time zone?

Reply:
Not sure what you mean. Could you provide me the code ?

Reply:
offset timezone is like `-07:00` instead of a geographical one like `America/Los_Angeles`